src/torchmetrics/detection/mean_ap.py

```diff
@@ -502,7 +502,7 @@ def __evaluate_image_gt_no_preds(
         return {
             "dtMatches": torch.zeros((nb_iou_thrs, nb_det), dtype=torch.bool, device=self.device),
             "gtMatches": torch.zeros((nb_iou_thrs, nb_gt), dtype=torch.bool, device=self.device),
-            "dtScores": torch.zeros(nb_det, dtype=torch.bool, device=self.device),
+            "dtScores": torch.zeros(nb_det, dtype=torch.float32, device=self.device),
```

Comment:
can we pls add a test to cover this case?

Reply:
@SkafteNicki I am currently unable to retrieve an input that causes the error but I think the change is fairly minute and straightforward. dtScores should be a tensor of type float32 according to documentation.

Reply:
test case added from this comment: https://github.com/Lightning-AI/metrics/issues/1147#issuecomment-1198910780